django/utils/_os.py

```diff
@@ -56,18 +56,17 @@ def symlinks_supported():
     host platform and/or if they are allowed to be created (e.g.
     on Windows it requires admin permissions).
     """
-    tmpdir = tempfile.mkdtemp()
-    original_path = os.path.join(tmpdir, 'original')
-    symlink_path = os.path.join(tmpdir, 'symlink')
-    os.makedirs(original_path)
-    try:
-        os.symlink(original_path, symlink_path)
-        supported = True
-    except (OSError, NotImplementedError, AttributeError):
-        supported = False
-    else:
-        os.remove(symlink_path)
-    finally:
-        os.rmdir(original_path)
-        os.rmdir(tmpdir)
-        return supported
+    with tempfile.TemporaryDirectory() as tmpdir:
+        original_path = os.path.join(tmpdir, 'original')
+        symlink_path = os.path.join(tmpdir, 'symlink')
+        os.makedirs(original_path)
+        try:
+            os.symlink(original_path, symlink_path)
+            supported = True
+        except (OSError, NotImplementedError, AttributeError):
```

Comment:
I think `AttributeError` can be removed as per comments removed in https://github.com/django/django/commit/1c466994d9729a8902d34e277a3a685bef2a08e1#diff-60e682a2651bd7c2b85cffd4a0cec874.